catalog/security/catalog-security-resourceplugin/src/main/java/org/codice/ddf/catalog/security/ResourceUriPolicy.java

```diff
@@ -0,0 +1,179 @@
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+import ddf.security.permission.Permissions;
+
+/**
+ * Restricts how resource URIs are updated and created.
+ * There are security risk to allowing users to update URIs as well as allowing users to
+ * pass in a URI when the metacard is created.
+ */
+public class ResourceUriPolicy implements PolicyPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceUriPolicy.class);
+
+    private String[] createPermissions;
+
+    private String[] updatePermissions;
+
+    public ResourceUriPolicy() {
+    }
+
+    public ResourceUriPolicy(String[] createPermissions, String[] updatePermissions) {
+        setCreatePermissions(createPermissions);
+        setUpdatePermissions(updatePermissions);
+    }
+
+    @Override
+    public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        if (!isLocalRequest(properties)) {
+            return new PolicyResponseImpl();
+        }
+
+        if (input.getResourceURI() != null && StringUtils.isNotEmpty(input.getResourceURI()
+                .toString())) {
+            return new PolicyResponseImpl(null,
+                    Permissions.parsePermissionsFromString(getCreatePermissions()));
+        }
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreUpdate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        if (!isLocalRequest(properties)) {
+            return new PolicyResponseImpl();
+        }
+
+        PolicyResponseImpl policyResponse = new PolicyResponseImpl(null,
+                Permissions.parsePermissionsFromString(getUpdatePermissions()));
+
+        List<Metacard> previousStateMetacards = ((OperationTransaction) properties.get(
+                OPERATION_TRANSACTION_KEY)).getPreviousStateMetacards();
+
+        Metacard previous;
+        try {
+            previous = previousStateMetacards.stream()
+                    .filter((x) -> x.getId()
+                            .equals(input.getId()))
+                    .findFirst()
+                    .get();
+        } catch (NoSuchElementException e) {
+            LOGGER.debug("Cannot locate metacard {} for update. Applying permissions to the item",
+                    input.getId());
+            return policyResponse;
+        }
+
+        return requiresPermission(input.getResourceURI(), previous.getResourceURI()) ?
+                policyResponse :
+                new PolicyResponseImpl();
+    }
+
+    private boolean isLocalRequest(Map<String, Serializable> properties) {
```

Comment:
Is there a reason you created this method instead of directly using `Requests.isLocal(Map<String, Serializable> props)`